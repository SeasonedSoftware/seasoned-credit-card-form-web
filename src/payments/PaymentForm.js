import React from 'react'
import { Formik } from 'formik'
import PropTypes from 'prop-types'

import FormContainer from './form/FormContainer'
import validate from './form/validate'

const PaymentForm = ({ onSubmit, ...props }) => {
  return (
    <Formik
      initialValues={{
        name: '',
        number: '',
        expiry: '',
        cvc: '',
      }}
      validate={validate(props)}
      onSubmit={onSubmit}
      render={options => <FormContainer {...props} {...options} />}
    />
  )
}

PaymentForm.defaultProps = {
  language: 'en-US',
}

PaymentForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  error: PropTypes.string,
  acceptedCards: PropTypes.arrayOf(
    PropTypes.oneOf([
      'visa',
      'amex',
      'elo',
      'mastercard',
      'diners',
      'hipercard',
    ]),
  ),
  language: PropTypes.oneOf(['en-US', 'pt-BR']),
  buttonMessage: PropTypes.string,
}

export default PaymentForm
