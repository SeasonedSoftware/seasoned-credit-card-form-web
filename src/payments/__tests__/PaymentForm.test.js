import React from 'react'
import renderer from 'react-test-renderer'

import PaymentForm from '../PaymentForm'

jest.mock('formik', () => ({
  Formik: ({ render, ...props }) => (
    <div {...props}>Formik - {render(props)}</div>
  ),
}))

jest.mock('../form/FormContainer', () => props => (
  <div {...props}>FormContainer</div>
))

it('renders correctly', () => {
  const props = {
    onSubmit: jest.fn(),
    foo: 'bar',
    submitting: false,
  }
  const tree = renderer.create(<PaymentForm {...props} />)
  expect(tree.toJSON()).toMatchSnapshot()
})
