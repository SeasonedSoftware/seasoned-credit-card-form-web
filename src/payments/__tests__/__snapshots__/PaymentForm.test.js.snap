// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  initialValues={
    Object {
      "cvc": "",
      "expiry": "",
      "name": "",
      "number": "",
    }
  }
  onSubmit={[MockFunction]}
  validate={[Function]}
>
  Formik - 
  <div
    foo="bar"
    initialValues={
      Object {
        "cvc": "",
        "expiry": "",
        "name": "",
        "number": "",
      }
    }
    language="en-US"
    onSubmit={[MockFunction]}
    submitting={false}
    validate={[Function]}
  >
    FormContainer
  </div>
</div>
`;
